### PR TITLE
[AL-4867] Create batch using global keys

### PR DESCRIPTION
**Pull request by whistler**
Depends on: https://github.com/Labelbox/intelligence/pull/13652

### CODE REVIEW

File: labelbox/schema/project.py [hunk 1]
```diff
@@ -688,35 +690,40 @@ def create_batch(self,
...
-            else:
-                raise ValueError("You can DataRow ids or DataRow objects")
+        if data_rows is not None:
+            for dr in data_rows:
+                if isinstance(dr, Entity.DataRow):
+                    dr_ids.append(dr.uid)
+                elif isinstance(dr, str):
+                    dr_ids.append(dr)
+                else:
+                    raise ValueError("You can DataRow ids or DataRow objects")
```
Comment (apollonin) [inline labelbox/schema/project.py:700]: I know it is unrelated to this PR, but should pretty easy to improve. The error text is weird and would be cool to add "provide" word in there =)

File: labelbox/schema/project.py [hunk 2]
```diff
@@ -1260,38 +1270,47 @@ def _is_url_valid(url: Union[str, Path]) -> bool:
...
+            queryAllDataRowsHaveBeenProcessed(dataRowIds:$%s, globalKeys:$%s) {
                 allDataRowsHaveBeenProcessed
            }
-        }""" % (data_row_ids_param, data_row_ids_param)
+        }""" % (data_row_ids_param, global_keys_param, data_row_ids_param,
+                global_keys_param)
 
-        params = {}
-        params[data_row_ids_param] = data_row_ids
+        params = {
```
Comment (vbrodsky) [inline labelbox/schema/project.py:1310]: do we have to send both if only one will be ever present?

File: labelbox/schema/project.py [hunk 3]
```diff
@@ -671,14 +671,16 @@ def setup(self, labeling_frontend, labeling_frontend_options) -> None:
 
     def create_batch(self,
                      name: str,
-                     data_rows: List[Union[str, DataRow]],
+                     data_rows: List[Union[str, DataRow]] = None,
+                     global_keys: List[str] = None,
                      priority: int = 5,
                      consensus_settings: Optional[Dict[str, float]] = None):
-        """Create a new batch for a project. Batches is in Beta and subject to change
+        """Create a new batch for a project. One of `global_keys` or `data_rows` must be provided but not both.
```
[the remark this answers is not in the document]
Comment (whistler) [reply labelbox/schema/project.py:678]: This is validated on the backend.

File: labelbox/schema/project.py (hunk 1 above)
Comment (whistler) [reply labelbox/schema/project.py]: Good catch, updated!

File: labelbox/schema/project.py (hunk 2 above)
Comment (whistler) [reply labelbox/schema/project.py:1315]: I'll see if I can pass just one.

Comment (whistler) [reply labelbox/schema/project.py]: Updated so that only one of globalKeys or dataRowIds is passed.

**Review by kkim-labelbox — APPROVED** [review]
LGTM!

**Comment by whistler** [comment]
The one test that doesn't pass is also broken on develop, so merging this.

**Pull request merged**